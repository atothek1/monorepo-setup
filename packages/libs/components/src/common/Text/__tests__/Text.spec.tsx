import React from "react";
import { render } from "@mono/testing";
import { Text } from "../index";

describe( "Text snapshots", () => {
    const testId = "testid.text";

    it( "should render default", () => {
        const { getByTestId } = render( <Text testId={ testId }>test</Text> );
        expect( getByTestId( testId ) ).toMatchSnapshot();
    } );

    it( "should render as p tag", () => {
        const { getByTestId } = render( <Text as="p" testId={ testId }>test</Text> );
        expect( getByTestId( testId ) ).toMatchSnapshot();
    } );

    it( "should render as h1 tag", () => {
        const { getByTestId } = render( <Text as="h1" testId={ testId }>test</Text> );
        expect( getByTestId( testId ) ).toMatchSnapshot();
    } );
} );
