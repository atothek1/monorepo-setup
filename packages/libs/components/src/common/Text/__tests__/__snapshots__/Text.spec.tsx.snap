// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text snapshots should render as h1 tag 1`] = `
.c0 {
  font-family: Roboto;
  font-weight: 400;
  color: #222222;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: 1.6rem;
  line-height: 1.5;
}

<h1
  class="Text___StyledStyledText-sc-1iaoujd-0 c0"
  data-testid="testid.text"
>
  test
</h1>
`;

exports[`Text snapshots should render as p tag 1`] = `
.c0 {
  font-family: Roboto;
  font-weight: 400;
  color: #222222;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: 1.6rem;
  line-height: 1.5;
}

<p
  class="Text___StyledStyledText-sc-1iaoujd-0 c0"
  data-testid="testid.text"
>
  test
</p>
`;

exports[`Text snapshots should render default 1`] = `
.c0 {
  font-family: Roboto;
  font-weight: 400;
  color: #222222;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-size: 1.6rem;
  line-height: 1.5;
}

<span
  class="Text___StyledStyledText-sc-1iaoujd-0 c0"
  data-testid="testid.text"
>
  test
</span>
`;
